core/block.ts

```diff
@@ -343,23 +343,17 @@ export class Block implements IASTNodeLocation, IDeletable {
       this.workspace.removeChangeListener(this.onchangeWrapper_);
     }
 
-    eventUtils.disable();
-    try {
-      this.workspace.removeTypedBlock(this);
-      this.workspace.removeBlockById(this.id);
-      this.disposing = true;
-
-      this.childBlocks_.forEach((c) => c.disposeInternal());
-      this.inputList.forEach((i) => i.dispose());
-      this.inputList.length = 0;
-      this.getConnections_(true).forEach((c) => c.dispose());
-    } finally {
-      eventUtils.enable();
```

Comment:
I don't think we need to disable events any longer since the delete event is fired by `dispose` and not `disposeInternal`, which is what we use to recurse.

Said another way, only the top level block fires a delete event, so we have no need to disable events for child blocks.

Here is the context for originally disabling events: https://github.com/google/blockly/commit/c429949a4e449fc84e46f79e8eba55f5df9c06cb